cypress-io/cypress
# chore(webkit): add video recording

- For #6422 

### User facing changelog

n/a

### Additional details

- Moves browser-specific video recording logic out of `run.ts` and in to the `browsers/*` drivers.
	- This happened in https://github.com/cypress-io/cypress/pull/23579/commits/e3b197998b34efc110a6ca087ada8beb0a0838a5
	- Browsers now control the video recording more closely, and only expose a small API to `run.ts` (start/end/restart control)
- Adds WebKit video recording.
	- This necessitated the above refactor - WebKit only supports recording an entire Page session at a time, and it streams to disk, so we can't use the ffmpeg method we regularly use with WK.
- Adds a `webkit-system-tests` job. 60 `.it`s are skipped in this PR since they don't work yet. It breaks down like this:
	- 6 skipped due to needing multidomain support
	- 8 skipped due to missing before:browser:launch support
	- 22 skipped due to broken stack traces
	- 24 other
- But importantly for this PR, the `video_compression` system-test spec is unskipped and passing in WebKit
- [x] Depends on #23587

### Steps to test

Run a test project that records video in WebKit, for example:

```
yarn cypress:run --project ../cypress-test-tiny --browser webkit --config video=true
```

(the project needs to have `playwright-webkit` installed)

Video should be recorded to the videosFolder after doing this. No guarantee that the project won't fail for other reasons since WebKit is WIP.

### How has the user experience changed?


### PR Tasks


- [na] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
